File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {

    static int N, M, V;
    static List<List<Integer>> graph = new ArrayList<>();
    static boolean[] visited;
    static BufferedWriter bw;

    static void dfs(int v) throws IOException {
        visited[v] = true;
        bw.write(v + " ");

        for (int w : graph.get(v)) {
            if (!visited[w]) {
                dfs(w);
            }
        }
    }

    static void bfs(int v) throws IOException {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(v);
        visited[v] = true;

        while (!queue.isEmpty()) {
            int cur = queue.poll();
            bw.write(cur + " ");

            for (int w : graph.get(cur)) {
                if (!visited[w]) {
                    visited[w] = true;
                    queue.add(w);
                }
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());

        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());
        visited = new boolean[N + 1];

        for (int i = 0; i <= N; i++) graph.add(new ArrayList<>());

        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph.get(a).add(b);
            graph.get(b).add(a);
        }

        for (int i = 1; i <= N; i++) {
            Collections.sort(graph.get(i));
        }

        dfs(V);
        bw.newLine();

        Arrays.fill(visited, false);
        bfs(V);
        bw.newLine();

        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque


def dfs(board, visited, v):
    stack = [v]
    while stack:
        v = stack.pop()
        if not visited[v]:
            visited[v] = True
            print(v, end=' ')
            for value in sorted(board[v], reverse=True):
                if not visited[value]:
                    stack.append(value)

def bfs(board, visited, v):
    queue = deque([v])
    while queue:
        v = queue.popleft()
        if not visited[v]:
            visited[v] = True
            print(v, end=' ')
            for value in sorted(board[v]):
                if not visited[value]:
                    queue.append(value)

n, m, v = map(int, input().split())

board = [[] for _ in range(n + 1)]

for i in range(m):
    a, b = map(int, input().split())
    board[a].append(b)
    board[b].append(a)

visited = [False] * (n + 1)

dfs(board, visited, v)
print()
visited = [False] * (n + 1)
bfs(board, visited, v)